qm-plugins/qm-cron/class-qm-cron-output-html.php

```diff
@@ -0,0 +1,333 @@
+			$show_time   = true;
+
+			foreach ( $time_cron_array as $hook => $data ) {
+				$row_attributes = $this->get_event_row_attributes( $time, $hook );
+				$arg_set_count  = count( $data );
+				$show_hook      = true;
+
+				foreach ( $data as $hash => $info ) {
+					echo '<tr', $row_attributes, '>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+
+					if ( true === $show_time ) {
+						$this->display_event_time( $time, $event_count );
+						$show_time = false;
+					}
+
+					if ( true === $show_hook ) {
+						$this->display_event_hook( $hook, $arg_set_count );
+						$show_hook = false;
+					}
+
+					// Report the schedule.
+					echo '<td>';
+					$this->display_event_schedule( $info );
+					echo '</td>';
+
+					// Report the interval.
+					echo '<td class="intervals">';
+					$this->display_event_intervals( $info );
+					echo '</td>';
+
+					// Report the args.
+					echo '<td>';
+					$this->display_event_cron_arguments( $info['args'] );
+					echo '</td></tr>';
+				}
+				unset( $hash, $info );
+			}
+			unset( $hook, $data, $row_attributes, $arg_set_count, $show_hook );
+		}
+		unset( $time, $time_cron_array, $show_time );
+
+		echo '</tbody></table>';
+	}
+
+	/**
+	 * Count the number of argument sets for a cron time.
+	 *
+	 * @param array $hook_array Array of hooks with argument sets.
+	 *
+	 * @return int
+	 */
+	private function get_arg_set_count( $hook_array ) {
+		$count = 0;
+		foreach ( $hook_array as $set ) {
+			$count += count( $set );
+		}
+		return $count;
+	}
+
+	/**
+	* Create a HTML attribute string for an event row.
+	*
+	* @param int    $time Unix timestamp.
+	* @param string $hook Action hook for the cron job.
+	*
+	* @return string
+	*/
+	private function get_event_row_attributes( $time, $hook ) {
+		$attributes = '';
+
+		// Verify if any events are hooked in.
+		if ( false === has_action( $hook ) ) {
+			/* translators: This text will display as a tooltip. %1$s will be replaced by a line break. */
+			$attributes .= ' title="' . sprintf( esc_attr__( 'No actions are hooked into this event at this time.%1$sThe most likely reason for this is that a plugin or theme was de-activated or uninstalled and didn\'t clean up after itself.%1$sHowever, a number of plugins also use the best practice of lean loading and only hook in conditionally, so check carefully if you intend to remove this event.', 'query-monitor' ), "\n" ) . '"';
+		}
+
+		return $attributes;
+	}
+
+	/**
+	* Display the timing for the event as a date, timestamp and human readable time difference.
+	*
+	* @param int $time        Timestamp.
+	* @param int $event_count Number of events running at this time.
+	*/
+	private function display_event_time( $time, $event_count ) {
+		$row_span = ( $event_count > 1 ) ? ' rowspan="' . esc_attr( $event_count ) . '"' : '';
+
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo '<td' . $row_span . '>
+		   ', esc_html( gmdate( 'Y-m-d H:i:s', $time ) ), '<br />
```

Comment:
We use self-closing `br` tags here; does it make sense to use them everywhere for consistency? For example, we use `<br>`, not `<br />` in `output()`.